source/dynamic_menu.sqf

```diff
@@ -0,0 +1,160 @@
+};
+
+// Compilaton and Call for Offensive SubMenu
+_finalString = "offensive_SubMenu = [[""Offensive Supports"",true],";
+_offensiveItemsClassString = _offensiveItems joinString ",";
+_finalString = _finalString + _offensiveItemsClassString;
+_finalString = _finalString + "];";
+
+_myCode = compile _finalString;
+call _myCode;
+
+
+//Logistics Submenu Checks
+if(support_supplydrop_available) then {
+    _text = ["Supply Drop (2 CP)", 2, '_null = [player] execVM ""support\ammobox.sqf""', 1] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+} else {
+    _text = ["Supply Drop ((N/A)", 2, '', 0] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+};
+
+if(support_paradrop_available) then {
+    _text = ["Airborne troops (25 CP)", 3, '_null = [player] execVM ""support\paradrop.sqf""', 1] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+} else {
+    _text = ["Airborne troops (N/A)", 3, '', 0] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+};
+
+if(support_uav_recon_available) then {
+    _text = ["UAV Recon (3 CP)", 4, '_null = [player] execVM ""support\mapclickuav.sqf""', 1] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+} else {
+    _text = ["UAV Recon (N/A)", 4, '', 0] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+};
+
+if(support_veh_refit_available) then {
+    _text = ["Vehicle Refit (2 CP)", 5, '_null = [player] execVM ""support\veh_refit.sqf""', 1] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+} else {
+    _text = ["Vehicle Refit (N/A)", 5, '', 0] call menuItemFormatter;
+    _logisticsItems pushBack _text;
+};
+
+
+// Compilaton and Call for Logistics SubMenu
+_finalString = "logistic_SubMenu = [[""Logistical Supports"",true],";
+_logisticsItemsClassString = _logisticsItems joinString ",";
+_finalString = _finalString + _logisticsItemsClassString;
+_finalString = _finalString + "];";
+
+_myCode = compile _finalString;
+call _myCode;
+
+//Transport Submenu check
+if(support_helotaxi_available) then {
+    _text = ["Helicopter taxi(1 CP)", 2, '_nill = [getpos player,50] execVM ""support\taxi\helotaxi.sqf""', 1] call menuItemFormatter;
+    _transportItems pushBack _text;
+} else {
+    _text = ["Helicopter taxi(N/A)", 2, '', 0] call menuItemFormatter;
+    _transportItems pushBack _text;
+};
+
+if(support_boattaxi_available) then {
+    _text = ["Boat taxi (1 CP)", 3, '_null = [getpos player,10] execVM ""support\taxi\boattaxi.sqf""', 1] call menuItemFormatter;
+    _transportItems pushBack _text;
+} else {
+    _text = ["Boat taxi (N/A)", 3, '', 0] call menuItemFormatter;
+    _transportItems pushBack _text;
+};
+
+
+// Compilaton and Call for Transport SubMenu
+_finalString = "transport_SubMenu = [[""Transport Supports"",true],";
+_transportItemsClassString = _transportItems joinString ",";
+_finalString = _finalString + _transportItemsClassString;
+_finalString = _finalString + "];";
+
+_myCode = compile _finalString;
```

Comment:
compiling scripts is very expensive.
To replace this you can use `setVariable` in combination with `parseSimpleArray`